### DISCUSSION TITLE
PascalCase support for search results

### DISCUSSION THREAD

**Discussion by wofsauge** [squidfunk/mkdocs-material#4427] [Q&A] [answered]
Hello,

my goal was to make the search feature interpret PascalCase and camelCase words, so that they can be correctly found via the search feature. For that usecase i refered to the fairly recent [blog post](https://squidfunk.github.io/mkdocs-material/blog/2021/09/13/search-better-faster-smaller/#tokenizer-lookahead) which provides a new search-seperator configuration thats supposed to support it. When its added to the docs however, it spits out a lot of false positives and not the results i hoped it would.

For example, i have a H3 headline that is called "AddCoins ()". As a user of the search feature, i want to get this headline listed as a result when i type in any of these searches:
- addcoins
- coins
- add coins

But as it is right now, only the first searchquery lists the headline as a result. The other only show the file that headline is in.

Additionally, it also randomly highlights any meantions of single characters (like "o", or "c") from the search query, making the results very confusing.

Does anyone know why these bugs occur and why the other search-queries not work?

Here is a minimal project with this bug as an example:
[minimal example docs.zip](https://github.com/squidfunk/mkdocs-material/files/9678957/minimal.example.docs.zip)

**Comment by squidfunk** [comment, marked as answer]
Thanks for providing the reproduction! Are you using [Insiders](https://squidfunk.github.io/mkdocs-material/insiders/)? The new search is currently [only available to sponsors](https://squidfunk.github.io/mkdocs-material/insiders/#how-to-become-a-sponsor) of the project. When I run your example with Insiders, it works as intended:

<img width="687" alt="Bildschirmfoto 2022-09-30 um 08 07 19" src="https://user-images.githubusercontent.com/932156/193201995-265aa0f5-3a33-4882-af8d-2db10b7f8ff3.png">

<img width="683" alt="Bildschirmfoto 2022-09-30 um 08 07 32" src="https://user-images.githubusercontent.com/932156/193202015-7c9c115c-e4f2-4419-8cc7-c22891c81bbe.png">

Note that searching for `addcoins` will not work as expected, because the query is essentially one word, and the tokenizer splits it into two in the search. You might switch to non-pascal case tokenization for that.
